bindings/python/src/lib.rs

```diff
@@ -300,17 +300,45 @@ fn find_cudart(module: &PyModule) -> Option<Library> {
     if var != "1" {
         return None;
     }
-    let path: std::path::PathBuf = module
+
+    let path: PyResult<std::path::PathBuf> = module
         .getattr(intern!(module.py(), "_C"))
         .ok()?
         .getattr(intern!(module.py(), "__file__"))
         .ok()?
-        .extract()
-        .ok()?;
-    // SAFETY: This is unsafe because the library might run arbitrary code
-    // So it's really important to make sure we are targeting the correct
-    // library.
-    let lib = unsafe { Library::new(path).ok()? };
+        .extract();
+
+    #[cfg(not(target_os = "windows"))]
+    let lib = {
+        let path = path.ok()?;
+        // SAFETY: This is unsafe because the library might run arbitrary code
+        // So it's really important to make sure we are targeting the correct
+        // library.
+        unsafe { Library::new(path).ok()? }
+    };
+
+    #[cfg(target_os = "windows")]
+    let lib = {
+        let mut path = path.ok()?;
+        path.pop();
+        path.push("lib");
+        let paths = std::fs::read_dir(path).ok()?;
+        let cudart_paths: Vec<_> = paths
+            .into_iter()
+            .filter_map(|p| {
+                let p = p.ok()?.path();
+                let filename = p.file_name()?.to_string_lossy();
+                if filename.starts_with("cudart") {
+                    Some(p)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        let cudart_path = cudart_paths.get(0)?;
+        Library::from(libloading::os::windows::Library::open_already_loaded(cudart_path).ok()?)
```

Comment:
As safe as the DLL named `cudart_path` is safe :smile:

Reply:
No, it won't load a DLL that isn't already loaded by the program ! So if someone puts a fake DLL it wouldn't get loaded (if I understand what I read, I didn't check it out myself)

Reply:
It does seem to limit the attack surface quite a bit, I'm unfamiliar with what "already loaded" mechanics mean though

Reply:
When you load a program that has linked dynamic libraries, the OS will go and resolve those DLLs at startup time. (It won't resolve individual symbols though so you might have a later error when accessing said functions).

Since `torch` uses `torch._C.so` which is a DLL which is linked against the aforementionned DLLs it will load `cudart-xxx.dll` too.

Reply:
Thanks for explaining !

Reply:
torch loads kernel32.dll and maps LoadLibraryEx and a couple other functions into torch.kernel32.LoadLibraryEx() in its namespace and leaves it there for the duration of its run on Windows. 

I'm not too worried about whatever other DLL gets loaded in comparison.   👯